samcli/local/docker/container.py

```diff
@@ -272,6 +274,31 @@ def start(self, input_data=None):
         # Start the container
         real_container.start()
 
+        # Wait for port to be open
+        self.wait_for_port()
+
+    def wait_for_port(self):
+        """
+        Waits until the host machine port that Docker binds to is open.
+        """
+        start_time = time.time()
+        sleep = 0.1
+        timeout = 10
+        while True:
+            a_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            location = (self._container_host_interface, self.rapid_port_host)
+            is_port_open = not a_socket.connect_ex(location)
+            a_socket.close()
+
+            if is_port_open:
+                break
+
+            current_time = time.time()
+            if current_time - start_time > timeout:
+                raise RuntimeError("Timed out while starting container")
```

Comment:
Is this the right exception to be using?

Reply:
May be use a `UserException` will be better, also you let the customer knows that increasing the timeout and retry may fix the issue (in case if you will make the timeout configurable)